Library/Formula/telegram.rb

```diff
@@ -0,0 +1,40 @@
+class Telegram < Formula
+  homepage "https://github.com/vysheng/tg"
+  url "https://github.com/vysheng/tg/archive/1.0.5.1.tar.gz"
+  sha1 "72f6a9f89ee9db90b573bad25e567a2015115848"
+
+  depends_on "openssl"
+  depends_on "libevent"
+  depends_on "libconfig"
+  depends_on "readline"
+  depends_on "lua"
+
+  patch :DATA
+
+  def install
+    ENV.append "CFLAGS", "-I#{HOMEBREW_PREFIX}/include -I#{Formula["readline"].prefix}/include"
+    ENV.append "LDFLAGS", "-L#{HOMEBREW_PREFIX}/lib -L#{Formula["readline"].prefix}/lib"
```

Comment:
We don't want to add `HOMEBREW_PREFIX` here, though, as that will pick up things that aren't depended on.